cloud/pkg/cloudhub/servers/udsserver/uds.go

```diff
@@ -54,11 +57,12 @@ func (us *UnixDomainSocket) StartServer() error {
 		return err
 	}
 	if proto == "unix" {
-		addr = "/" + addr
-		if err := os.Remove(addr); err != nil && !os.IsNotExist(err) { //nolint: vetshadow
-			klog.Errorf("failed to remove addr: %v", err)
+		addr = filepath.Join("/" + addr)
+		if err := checkUnixSocket(addr); err != nil {
+			klog.Errorf("failed to check unix socket addr: %v", err)
 			return err
 		}
+		defer os.Remove(addr)
 	}
```

Comment:
i think don't need this,  function `checkUnixSocket` already remove socket. or we can remove `os.Remove` from `checkUniSocket` and exec `os.Remove` out of `checkUniSocket` and return error to `StartServer`

Reply:
This is to ensure that the created sockets are removed when the unix server exits，although for the time being it does not appear that the `StartServer` will exit gracefully(TODO)

It is not necessary to have a socket when the program is not running.

Reply:
got it, thanks